src/shogun/lib/SGMatrix.cpp

```diff
@@ -1022,6 +1031,7 @@ SGMatrix<T> SGMatrix<T>::get_allocated_matrix(index_t num_rows,
 template<class T>
 void SGMatrix<T>::copy_data(const SGReferencedData &orig)
 {
+	gpu_ptr=std::shared_ptr<GPUMemoryBase<T>>(((SGMatrix*)(&orig))->gpu_ptr);
```

Comment:
creating a new shared_ptr? not sure about this. why not just directly assign the orig->gpu_ptr?